src/robot/libraries/String.py

```diff
@@ -121,6 +123,47 @@ def decode_bytes_to_string(self, bytes, encoding, errors='strict'):
             raise TypeError('Can not decode strings on Python 3.')
         return bytes.decode(encoding, errors)
 
+    def format_string(self, template_or_str, *positional_search_replace,
+                      **named_search_replace):
+        """Formats a ``string`` using the given ``positional_search_replace``
+        and ``named_search_replace``.
+
+        If the given ``string`` is a valid absolute file path, opens the file in read
+        mode and then format its content using the given
+        ``positional_search_replace`` and ``named_search_replace``.
+
+        This keyword uses python's string format. For more information see:
+        [https://docs.python.org/library/string.html#formatstrings]|Format syntax]
+
+        Examples:
+
+        `Considering the file C:\\template.txt contents as "My {test} String"`
+        | ${result} = | Format String | C:\\template.txt | | test=awesome |
+        | Should Be Equal | ${result} | My awesome String |
+        | ${result} = | Format String | User {} is not a admin user. | non-admin | |
+        | Should Be Equal | ${result} | User non-admin is not a admin user. |
+        | ${result} = | Format String | Username: {username} - Password: {password} | | username=Robot | password=Framework |
+        | Should Be Equal | ${result} | Username: Robot - Password: Framework |
+        | ${result} = | Format String | Document {} is missing on folder {folder} | tests.robot | folder=/home |
+        | Should Be Equal | ${result} | Document tests.robot is missing on folder /home |
+        | ${result} = | Format String | Uploaded file: {} should not be bigger than {}. | photo.jpg | 5MB |
+        | Should Be Equal | ${result} | Uploaded file: photo.jpg should not be bigger than 5MB. |
+
+        New in Robot Framework 3.1.
+        """
+        if isabs(template_or_str) and isfile(template_or_str):
+            logger.info(
+                'Reading template from file <a href="%s">%s</a>' % (template_or_str, template_or_str),
+                html=True)
+            template_or_str.replace("/", sep)
+            with open(template_or_str, 'r') as format_file:
+                format_string = format_file.read().rstrip()
```

Comment:
1. I agree it's a good idea to use `rstrip()`, but stripping trailing whitespace should be mentioned in the documentation as well. Obviously stripping is annoying if you want to have whitespace at the end, but leaving the newline that many editors automatically add to the end of a file would be annoying as well. One possibility would be stripping only possible newline. Do you thin that would be worth the (small) effor?

2. The current code won't work with Python 2 if the file contains non-ASCII characters and with Python 3 encoding would be operating system specific. We should decode the contents using UTF-8 like we do with all other data files. The easiest way to do that in Python 2 and 3 compatible manner is using our `robot.utils.UtfReader` like

```python
from robot.utils import Utf8Reader

with Utf8Reader(template) as reader:
    content = reader.read().rstrip()
```

3. Above reminds some new tests are needed:
   - Stripping whitespace from template files
   - Non-ASCII content in template files
   - Non-ASCII template strings in general

Reply:
1. We can acomplhis it with rstrip, I just need to pass the arg **'\r\n'** and it would only remove those characters.
2. Changed to use this reader
3. Added the new suggested tests

Reply:
> Obviously stripping is annoying if you want to have whitespace at the end, but leaving the newline that many editors automatically add to the end of a file would be annoying as well. 

What's the use case of stripping the end newlines? Having `\n` as the last byte of a file is a good practice - pep8 requires it through W292, diff complains if there isn't one, it's required in c/c++ header files (due to the `#include` directive), and running such file through `cat` is unpleasant as 
it leaves the prompt at the line's end, to name a few examples.

If we include it here, this will turn the keyword from a formatter to a mutator; for some users the trailing newlines may be significant. Also this adds behavior duality - the chars at stripped in file content, but not when a string is passed.

Reply:
All right, I'm removing it. But I really think people are going to have issues with new line from editors. I'm going to also add it to the documentation.